prow/entrypoint/run.go

```diff
@@ -83,10 +82,20 @@ func (o Options) Run() error {
 		commandErr = err
 	case <-time.After(timeout):
 		logrus.Errorf("Process did not finish before %s timeout", timeout)
-		if err := command.Process.Kill(); err != nil {
-			logrus.WithError(err).Error("Could not kill process after timeout")
+		if err := command.Process.Signal(os.Interrupt); err != nil {
+			logrus.WithError(err).Error("Could not interrupt process after timeout")
+		}
+		gracePeriod := time.Duration(optionOrDefault(o.GracePeriodSeconds, DefaultGracePeriodSeconds)) * time.Second
+		select {
+		case err := <-done:
+			commandErr = err
```

Comment:
This means we will show the error that the script generates on `SIGTERM` instead of a timeout message, which is probably fine?

Reply:
Could we make sure it is not nil? IMO we should ensure the job fails if it times out